odoo/api.py

```diff
@@ -34,23 +36,60 @@
 from .tools.translate import _
 from odoo.tools.misc import StackMap
 
-import typing
 if typing.TYPE_CHECKING:
     from collections.abc import Callable
-    from odoo.sql_db import BaseCursor
     from odoo.models import BaseModel
     try:
         from typing_extensions import Self  # noqa: F401
     except ImportError:
         from typing import Self  # noqa: F401
-    M = typing.TypeVar("M", bound=BaseModel)
-else:
-    Self = None
-    M = typing.TypeVar("M")
+
+    # not sure Self can work outside of a class?
+    # maybe S and R should be the same type, are there cases where they differ?
+    S = typing.TypeVar("S", bound=BaseModel)
+    R = typing.TypeVar("R", bound=BaseModel)
+    CreateCaller = Callable[[S, 'ValuesType' | list['ValuesType']], R]
```

Comment:
> > Most of the time, we accept any sequence, not only lists.
> 
> BaseModel.create is defined as accepting a list.

Them, you can leave a list. Maybe we will review the signature in the future.

> > For the question above, you can simplify and use just one type var
> 
> T is a terrible name for this typevar, and furthermore incorrect as it's not bound.

I meant that you should use the same variable. Binding here optional because the decorator does not care about the binding - you can still leave it for documentation.

> > For the caller, I would just put the signature once on the function, it's easier to read than to navigate through the file.
> 
> `CreateCaller` is used 6 different times. Do ou mean callee, which are still used twice each?

`model_create_single` will soon be removed, so you will have less usages. It is easier to read the definition on the method directly instead of looking for the definition.